keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt

```diff
@@ -75,6 +75,17 @@ interface ResourceHandler<S : ResourceSpec, R : Any> : SpinnakerExtensionPoint {
    */
   suspend fun current(resource: Resource<S>): R?
 
+  /**
+   * Evaluates the resource diff and determines whether it can take action to resolve the diff.
+   * @return a response indicating if it can take action, and a message explaining why if it can't.
+   *
+   * The default value is to take action on all changes. Only override if needed.
+   */
+  suspend fun willTakeAction(
+    resource: Resource<S>,
+    resourceDiff: ResourceDiff<R>
+  ): ActionDecision = ActionDecision()
```

Comment:
I suggest making it explicit that the default is to take action here so you don't have to look at the other class:
```suggestion
  ): ActionDecision = ActionDecision(willAct = true)
```